ambry-router/src/main/java/com/github/ambry/router/CompressionMetrics.java

```diff
@@ -0,0 +1,141 @@
+package com.github.ambry.router;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.compression.Compression;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CompressionMetrics {
+  public static class AlgorithmMetrics {
+
+    // Algorithm's compress() metrics.
+    public final Meter compressRate;      // Rate compress() is called.
+    public final Counter compressError;   // Rate compress() throws exception.
+    public final Histogram compressRatioPercent;    // compression ratio % = 100 * UncompressedSize/CompressedSize
+    public final Histogram fullSizeCompressTimeInMicroseconds;
+    public final Histogram fullSizeCompressSpeedMBPerSec;
+    public final Histogram partialFullCompressTimeInMicroseconds;
+    public final Histogram partialFullCompressSpeedMBPerSec;
```

Comment:
what is partialFull ?

Reply:
4MB is full size.  < 4MB is partial full.  The metrics for < 4 MB is highly skewed due to overheads.  For compressing a 1KB text is probably less efficient than compressing 100 KB, so the metrics for compressing 1 KB will generate many misleading metrics (like low compression ratio, low throughput (due to overhead)).  So looking at the 4MB compression metrics provides a more consistent and comparable picture of the algorithm performance.  At first, I thought about emitting metrics for 4 MB chunks only, but that may miss issues when the chunk size is < 4 MB.  That's why I separated the metrics for full chunks from the partial full chunks.

Reply:
I see. Please put this in comments, otherwise no one would guess this.
Also consider an alternate name such as fullChunkCompressTime and subChunkCompressTime.
The word chunk rings a bell for 4MB.

Reply:
we can categorize blobs based on size and then create speed metrics for those categories. something like smallBlobCompressSpeed, mediumBlobCompressSpeed and largeBlobCompressionSpeed. Please search ServerMetrics class, there is a similar settings in that class.

Reply:
I don't think it's necessary to break it to multiple categories.  I believe the full size and partial-full size is good enough.  The goal of these metrics aren't for analysis.  We just want to make sure the algorithm isn't acting crazy.  So most of the time we'll just keep an eye on the time of the full size chunks.